src/cli/self_update/windows.rs

```diff
@@ -505,83 +512,111 @@ fn _apply_new_path(new_path: Option<Vec<u16>>) -> Result<()> {
+            Ok(Some(String::new()))
         }
-        Err(ref e) if e.kind() == io::ErrorKind::NotFound => Ok(Some(Vec::new())),
         Err(e) => Err(e).context(CLIError::WindowsUninstallMadness),
     }
 }
 
 // Returns None if the existing old_path does not need changing, otherwise
 // prepends the path_str to old_path, handling empty old_path appropriately.
-fn _add_to_path(old_path: Vec<u16>, path_str: Vec<u16>) -> Option<Vec<u16>> {
+fn _add_to_path(old_path: String, path_str: String) -> Option<String> {
     if old_path.is_empty() {
         Some(path_str)
     } else if old_path
-        .windows(path_str.len())
-        .any(|path| path == path_str)
+        // .windows(path_str.len())
+        // .any(|path| path == path_str)
+        .contains(&path_str)
     {
         None
     } else {
         let mut new_path = path_str;
-        new_path.push(b';' as u16);
-        new_path.extend_from_slice(&old_path);
+        new_path.push(';');
+        new_path += &old_path;
         Some(new_path)
     }
 }
 
+// // Returns None if the existing old_path does not need changing
+// fn _remove_from_path(old_path: Vec<u16>, path_str: Vec<u16>) -> Option<Vec<u16>> {
+//     let idx = old_path
+//         .windows(path_str.len())
+//         .position(|path| path == path_str)?;
+//     // If there's a trailing semicolon (likely, since we probably added one
+//     // during install), include that in the substring to remove. We don't search
+//     // for that to find the string, because if it's the last string in the path,
+//     // there may not be.
+//     let mut len = path_str.len();
+//     if old_path.get(idx + path_str.len()) == Some(&(b';' as u16)) {
+//         len += 1;
+//     }
+
+//     let mut new_path = old_path[..idx].to_owned();
+//     new_path.extend_from_slice(&old_path[idx + len..]);
+//     // Don't leave a trailing ; though, we don't want an empty string in the
+//     // path.
+//     if new_path.last() == Some(&(b';' as u16)) {
+//         new_path.pop();
+//     }
+//     Some(new_path)
+// }
+
 // Returns None if the existing old_path does not need changing
-fn _remove_from_path(old_path: Vec<u16>, path_str: Vec<u16>) -> Option<Vec<u16>> {
+fn _remove_from_path(old_path: String, path_str: String) -> Option<String> {
     let idx = old_path
-        .windows(path_str.len())
-        .position(|path| path == path_str)?;
+        // .windows(path_str.len())
+        // .position(|path| path == path_str)?;
+        .find(&path_str)?;
     // If there's a trailing semicolon (likely, since we probably added one
     // during install), include that in the substring to remove. We don't search
     // for that to find the string, because if it's the last string in the path,
     // there may not be.
     let mut len = path_str.len();
-    if old_path.get(idx + path_str.len()) == Some(&(b';' as u16)) {
+    if old_path.chars().collect::<Vec<char>>()[idx + path_str.len() - 1] == ';' {
```

Comment:
Why are you using `.chars()` here? The previous `.find()` returns a byte index, not a char index, so indexing like this makes no sense.

Normally in this case, you should use [`.get()`](https://doc.rust-lang.org/std/primitive.str.html#method.get).  However, since you're checking for `';'`, I think you can take a shortcut here by calling `.as_bytes()` followed by a `.get()`, and finally compare it with `Some(b';')`.